src/Core/src/VisualDiagnostics/Extensions/VisualDiagnosticsExtensions.Android.cs

```diff
@@ -0,0 +1,162 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using Android.Graphics;
+using Android.Views;
+using GL = Android.Opengl;
+
+namespace Microsoft.Maui
+{
+	public static class VisualDiagnosticsAndroidExtensions
+	{
+		internal static ViewTransform? GetViewTransform(this IView view)
+		{
+			var nativeView = view.GetNative(true);
+			if (nativeView == null)
+				return null;
+			return GetViewTransform(nativeView);
+		}
+
+		internal static ViewTransform? GetViewTransform(View view)
+		{
+			if (view == null || view.Matrix == null || view.Matrix.IsIdentity)
+				return null;
+
+			var m = new float[16];
+			var v = new float[16];
+			var r = new float[16];
+
+			GL.Matrix.SetIdentityM(r, 0);
+			GL.Matrix.SetIdentityM(v, 0);
+			GL.Matrix.SetIdentityM(m, 0);
```

Comment:
We should probably use the `System.Numerics.Matrix4x4` that Maui.Graphics pulls in. This way, we are not doing all this jni. Also, this can be used on all the platforms.

Reply:
I think I'm gonna get rid of the ViewTransform stuff for now. Looking at this and the other implementations, there are more syncing that could be done for these APIs to make better use of these tools, and it's outside of scope for what I need for these overlays.